src/main/java/io/xpring/xrpl/model/XrpDepositPreauth.java

```diff
@@ -0,0 +1,80 @@
+package io.xpring.xrpl.model;
+
+import io.xpring.common.XrplNetwork;
+import io.xpring.xrpl.ClassicAddress;
+import io.xpring.xrpl.ImmutableClassicAddress;
+import io.xpring.xrpl.Utils;
+import org.immutables.value.Value;
+import org.xrpl.rpc.v1.DepositPreauth;
+
+import java.util.Optional;
+
+/**
+ * Represents a DepositPreauth transaction on the XRP Ledger.
+ * <p>
+ * A DepositPreauth transaction gives another account pre-approval to deliver payments to the sender of this
+ * transaction.
+ * This is only useful if the sender of this transaction is using (or plans to use) Deposit Authorization.
+ *
+ * Note: authorize and unauthorize are mutually exclusive fields: one but not both should be set.
+ * Addresses are encoded as X-addresses.  See https://xrpaddress.info/.
+ * </p>
+ *
+ * @see "https://xrpl.org/depositpreauth.html"
+ */
+@Value.Immutable
+public interface XrpDepositPreauth {
+  static ImmutableXrpDepositPreauth.Builder builder() {
+    return ImmutableXrpDepositPreauth.builder();
+  }
+
+  /**
+   * (Optional) The XRP Ledger address of the sender to preauthorize, encoded as an X-address.
+   *
+   * @return A {@link String} representing the XRP Ledger address of the sender to preauthorize, encoded as an
+   *         X-address.
+   */
+  Optional<String> authorizeXAddress();
+
+  /**
+   * (Optional) The XRP Ledger address of a sender whose preauthorization should be revoked, encoded as an X-address.
+   *
+   * @return A {@link String} representing the XRP Ledger address of a sender whose preauthorization should be revoked,
+   *         encoded as an X-address.
+   */
+  Optional<String> unauthorizeXAddress();
+
+  /**
+   * Constructs an XrpDepositPreauth from a DepositPreauth protocol buffer.
+   *
+   * @param depositPreauth A {@link DepositPreauth} (protobuf object) whose field values will be used to construct an
+   *                       XrpDepositPreauth
+   * @param xrplNetwork The network that this transaction should occur on.
+   * @return An XrpDepositPreauth with its fields set via the analogous protobuf fields.
+   * @see "<https://github.com/ripple/rippled/blob/3d86b49dae8173344b39deb75e53170a9b6c5284/src/ripple/proto/org/xrpl/rpc/v1/transaction.proto#L159"
+   */
+  static XrpDepositPreauth from(DepositPreauth depositPreauth, XrplNetwork xrplNetwork) {
+    final boolean isTestNetwork = xrplNetwork == XrplNetwork.TEST || xrplNetwork == XrplNetwork.DEV;
+
+    Optional<String> authorizeXAddress = Optional.empty();
+    Optional<String> unauthorizeXAddress = Optional.empty();
+    if (depositPreauth.hasAuthorize()) {
+      ClassicAddress authorizeClassicAddress = ImmutableClassicAddress.builder()
+          .address(depositPreauth.getAuthorize().getValue().getAddress())
```

Comment:
i copied https://github.com/xpring-eng/Xpring-JS/pull/493 here, but...guessing it makes sense that we don't use a destination tag here?

Reply:
I believe that we can't authorize / unauthorize at tag fields (and in fact, the tag field doesn't even exist in this transaction type).

Reply:
yeah I think this is correct... makes me question whether it makes sense to use an X-address here but I suppose consistency in the SDK is worth it

Reply:
i think consistency is good here!